### ISSUE TITLE
scan: Undefined method `expression` on `nil`

### ISSUE THREAD

**Issue by Mange** [castwide/solargraph#221]
Minimal reproduction I've found:

```ruby
# frozen_string_literal: true

class Foo
  def bar
    @bar =
      if rand < 0.5
        1
      end
  end
end
```

```
› solargraph scan
Error testing Foo#bar | @bar (/home/mange/tmp/solargraph-test/example.rb 4) at /home/mange/tmp/solargraph-test/example.rb:5
[NoMethodError]: undefined method `expression' for nil:NilClass
solargraph-0.35.2/lib/solargraph/pin/base_variable.rb:46:in `block in probe'
solargraph-0.35.2/lib/solargraph/pin/base_variable.rb:45:in `each'
solargraph-0.35.2/lib/solargraph/pin/base_variable.rb:45:in `probe'
solargraph-0.35.2/lib/solargraph/shell.rb:170:in `block (2 levels) in scan'
solargraph-0.35.2/lib/solargraph/shell.rb:166:in `each'
solargraph-0.35.2/lib/solargraph/shell.rb:166:in `block in scan'
lib/ruby/2.4.0/benchmark.rb:293:in `measure'
solargraph-0.35.2/lib/solargraph/shell.rb:164:in `scan'
thor-0.20.3/lib/thor/command.rb:27:in `run'
thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
thor-0.20.3/lib/thor.rb:387:in `dispatch'
thor-0.20.3/lib/thor/base.rb:466:in `start'
solargraph-0.35.2/bin/solargraph:5:in `<top (required)>'
bin/solargraph:23:in `load'
bin/solargraph:23:in `<main>'
bin/ruby_executable_hooks:24:in `eval'
bin/ruby_executable_hooks:24:in `<main>'
› solargraph --version
0.35.2
```

It seems to be assignment to the result of an `if` being the problem. I've found multiple variants of this problem in some of my codebases.

Changing to this equivalent code fixes the issue:

```ruby
# frozen_string_literal: true

class Foo
  def bar
    if rand < 0.5
      @bar = 1
    else
      @bar = nil
    end
  end
end
```

**Comment by castwide** [comment]
Confirmed. Thanks for tracking it down. I'll work on a fix for the next release.

**Comment by castwide** [comment]
This was a [simple fix](https://github.com/castwide/solargraph/commit/c75ae7dffe28f859733e711e3ea62bb8b4a7c59d) and I had already planned an update today, so I went ahead an published it in v0.36.0.

**Comment by Mange** [comment]
Thanks! :heart:
